pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -444,6 +445,65 @@ func getAddressByName(client *gophercloud.ServiceClient, name types.NodeName) (s
 	return addrs[0].Address, nil
 }
 
+// getInterfacesByName returns the node interfaces of the specified instance.
+func getInterfacesByName(client *gophercloud.ServiceClient, name types.NodeName) ([]attachinterfaces.Interface, error) {
+	var interfaces []attachinterfaces.Interface
+
+	// get server
+	srv, err := getServerByName(client, name)
```

Comment:
This function combines two server lookups (`getServerByName` and `attachinterfaces.List`).  Simplify it to only do one (the latter), and take the server ID as a parameter (rather than `NodeName`) - this will make it easier to compose with other functions later.  (ie: change this to `getAttachedInterfacesByID`)